app/portainer/components/form-components/FileUpload/FileUploadField.tsx

```diff
@@ -0,0 +1,69 @@
+import { ChangeEvent, createRef } from 'react';
+
+import { r2a } from '@/react-tools/react2angular';
+import { Button } from '@/portainer/components/Button';
+
+import styles from './FileUploadField.module.css';
+
+export interface Props {
+  onChange(value: File): void;
+    value?: File;
+  title?: string;
+  required?: boolean;
+}
+
+export function FileUploadField({
+  onChange,
+  value,
+  title = 'Select a file',
+  required = false,
+}: Props) {
+
+  const fileRef = createRef<HTMLInputElement>();
+
+  return (
+    <div className="file-upload-field">
```

Comment:
see my comment about label in test. this can be a label element, then findByLabel will give you the input

Reply:
had to add the `aria-label` prop to the input for it to work, no need to change this to a label